bigtable/src/main/scala/magnolify/bigtable/BigtableType.scala

```diff
@@ -14,88 +14,130 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
 package magnolify.bigtable
 
 import java.nio.ByteBuffer
 
-import com.google.protobuf.ByteString
-import magnolia._
-
-import scala.collection._
 import com.google.bigtable.v2.Mutation
 import com.google.bigtable.v2.Mutation.SetCell
 import com.google.cloud.bigtable.data.v2.models.{Row, RowCell}
+import com.google.protobuf.ByteString
+import magnolia._
+import magnolify.shared.Converter
+import magnolify.shims.JavaConverters._
 
 import scala.annotation.implicitNotFound
+import scala.collection._
 import scala.language.experimental.macros
-import magnolify.shims.JavaConverters._
-
-sealed trait BigtableType[T] extends Serializable {
-
-  def from(v: Row, columnFamily: String = BigtableType.DEFAULT_COLUMN_FAMILY_NAME): T
 
-  def to(v: T, columnFamily: String = BigtableType.DEFAULT_COLUMN_FAMILY_NAME,
-               timestamp: Long = 0L,
-               columnQualifier: String = null): Iterable[Mutation]
+sealed trait BigtableType[T] extends Converter[T, Row, Seq[Mutation]] {
+  protected val columnFamily: String
+  protected val timestampMicros: Long
 
+  def apply(v: Row): T = from(v)
+  def apply(v: T): Seq[Mutation] = to(v)
 }
 
 object BigtableType {
-  val DEFAULT_COLUMN_QUALIFIER_NAME = "q"
   val DEFAULT_COLUMN_FAMILY_NAME = "f"
+  val DEFAULT_COLUMN_QUALIFIER_NAME = "q"
 
-  implicit def apply[T](implicit f: BigtableField[T]): BigtableType[T] = new BigtableType[T] {
+  implicit def apply[T](implicit f: BigtableField[T]): BigtableType[T] =
+    BigtableType(BigtableType.DEFAULT_COLUMN_FAMILY_NAME, 0L)
 
-    def from(v: Row, columnFamily: String = BigtableType.DEFAULT_COLUMN_FAMILY_NAME): T =
-      f.get(columnFamily, v, null)
+  def apply[T](_columnFamily: String, _timestampMicros: Long)(
+    implicit f: BigtableField[T]
+  ): BigtableType[T] = new BigtableType[T] {
+    override protected val columnFamily: String = _columnFamily
+    override protected val timestampMicros: Long = _timestampMicros
 
-    def to(v: T, columnFamily: String = BigtableType.DEFAULT_COLUMN_FAMILY_NAME,
-                 timestamp: Long = 0L,
-                 columnQualifier: String = null): Iterable[Mutation] =
-      f.put(columnQualifier, v).map(Mutations.newSetCellMutation(columnFamily, timestamp))
+    override def from(v: Row): T =
+      f.get(columnFamily, v, null)
 
+    override def to(v: T): Seq[Mutation] =
+      f.put(null, v).map { b =>
+        Mutation
+          .newBuilder()
+          .setSetCell(
+            b.setFamilyName(columnFamily).setTimestampMicros(timestampMicros)
+          )
+          .build()
+      }
   }
+
+  def mutationsToRow(key: ByteString, mutations: Seq[Mutation]): Row =
```

Comment:
Lifing `Seq[Mutation]` <=> `Row` conversion methods here since they're pretty handy? Also make mutations a `Seq` to avoid `Iterable` <=> `List` copies.